internal/errors/timeout.go

```diff
@@ -33,25 +33,72 @@ type TimeoutError interface {
 	timeoutError()
 }
 
-type timeoutError struct {
+// handlerTimeoutError represents a timeout on the handler side.
+type handlerTimeoutError struct {
+	Caller    string
 	Service   string
 	Procedure string
 	Duration  time.Duration
 }
 
-// NewTimeoutError constructs an instance of a TimeoutError for the given
-// service, procedure, and duration waited.
-func NewTimeoutError(Service string, Procedure string, Duration time.Duration) TimeoutError {
-	return timeoutError{
+var _ TimeoutError = handlerTimeoutError{}
+var _ HandlerError = handlerTimeoutError{}
+
+// HandlerTimeoutError constructs an instance of a TimeoutError representing
+// a timeout that occurred during the handler execution, with the caller,
+// service, procedure and duration waited.
+func HandlerTimeoutError(Caller string, Service string, Procedure string, Duration time.Duration) error {
+	return handlerTimeoutError{
+		Caller:    Caller,
+		Service:   Service,
+		Procedure: Procedure,
+		Duration:  Duration,
+	}
+}
+
+func (handlerTimeoutError) timeoutError() {}
+func (handlerTimeoutError) handlerError() {}
+
+func (e handlerTimeoutError) Error() string {
+	return fmt.Sprintf(`call to procedure %q of service %q from caller %q timed out after %v`,
```

Comment:
Ok, note that this error can only be created on the handler side. It is then prefixed by "remote timeout:" on the client side. So it will look like "remote timeout: Timeout: blabla" now.